x-pack/filebeat/input/cel/integration/integration_test.go

```diff
@@ -0,0 +1,506 @@
+					"protocol":              outputProtocol,
+					"enabled":               true,
+					"ssl.verification_mode": "none",
+				}),
+			},
+		},
+		{
+			Id:             "input-unit-1",
+			Type:           proto.UnitType_INPUT,
+			ConfigStateIdx: 0,
+			State:          proto.State_HEALTHY,
+			LogLevel:       proto.UnitLogLevel_DEBUG,
+			Config: &proto.UnitExpectedConfig{
+				Id:   "cel-cel-1e8b33de-d54a-45cd-90da-23ed71c482e5",
+				Type: "cel",
+				Name: "cel-1",
+				Source: integration.RequireNewStruct(t, map[string]interface{}{
+					"use_output": "default",
+					"revision":   0,
+				}),
+				DataStream: &proto.DataStream{
+					Namespace: "default",
+				},
+				Meta: &proto.Meta{
+					Package: &proto.Package{
+						Name:    "cel",
+						Version: "1.9.0",
+					},
+				},
+				Streams: []*proto.Stream{
+					{
+						Id: "cel-cel.cel-1e8b33de-d54a-45cd-90da-23ed71c482e2",
+						DataStream: &proto.DataStream{
+							Dataset: "cel.cel",
+						},
+						Source: integration.RequireNewStruct(t, map[string]interface{}{
+							"interval":                        "10s",
+							"program":                         `bytes(get(state.url).Body).as(body,{"events":[body.decode_json()]})`,
+							"redact.delete":                   false,
+							"regexp":                          nil,
+							"resource.url":                    svrOne.URL,
+							"publisher_pipeline.disable_host": true,
+						}),
+					},
+				},
+			},
+		},
+	}
+
+	noStream := []*proto.UnitExpected{
+		{
+			Id:             "output-unit",
+			Type:           proto.UnitType_OUTPUT,
+			ConfigStateIdx: 0,
+			State:          proto.State_HEALTHY,
+			LogLevel:       proto.UnitLogLevel_INFO,
+			Config: &proto.UnitExpectedConfig{
+				Id:   "default",
+				Type: "elasticsearch",
+				Name: "elasticsearch",
+				Source: integration.RequireNewStruct(t, map[string]interface{}{
+					"type":                  "elasticsearch",
+					"hosts":                 outputHosts,
+					"username":              outputUsername,
+					"password":              outputPassword,
+					"protocol":              outputProtocol,
+					"enabled":               true,
+					"ssl.verification_mode": "none",
+				}),
+			},
+		},
+	}
+
+	observedStates := make(chan *proto.CheckinObserved)
+	expectedUnits := make(chan []*proto.UnitExpected)
+	done := make(chan struct{})
+	server := &mock.StubServerV2{
+		CheckinV2Impl: func(observed *proto.CheckinObserved) *proto.CheckinExpected {
+			select {
+			case observedStates <- observed:
+				return &proto.CheckinExpected{
+					Units: <-expectedUnits,
+				}
+			case <-done:
+				return nil
+			}
+		},
+		ActionImpl: func(response *proto.ActionResponse) error {
+			return nil
+		},
+	}
+	defer server.Stop()
+
+	require.NoError(t, server.Start())
+
+	os.Args = []string{
+		"filebeat",
+		"-E", fmt.Sprintf(`management.insecure_grpc_url_for_testing="localhost:%d"`, server.Port),
+		"-E", "management.enabled=true",
+		"-E", "management.restart_on_output_change=true",
+	}
```

Comment:
What is happening here?

Reply:
So this is a necessary evil as I invoke filebeat in this fashion `filebeat.Filebeat()` and hence I need the os.Args to have certain values. On contrast, other integration tests invoke filebeat as an external process but I prefer my way as I can have breakpoints in beats code during the integration test. To this end, I add a defer func to revert to the initail os.Args after this test exits. Feel free to un-resolve this if you want to change the filebeat invocation as an external process